packages/millicast-viewer-demo/src/viewer.js

```diff
@@ -71,6 +70,21 @@ const newViewer = () => {
     addStream(event.streams[0]);
   });
 
+  millicastView.on('onMetadata', (event) => {
+    const decoder = new TextDecoder()
+    const metadata = event.metadata
+    metadata.track = event.track
+    const uuid = metadata.uuid
+    metadata.uuid = uuid.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '')
+    metadata.uuid = metadata.uuid.replace(/(.{8})(.{4})(.{4})(.{4})(.{12})/, "$1-$2-$3-$4-$5")
+    if (metadata.timecode) {
+      metadata.timecode = decoder.decode(metadata.timecode)
+    } else if (metadata.unregistered) {
+      metadata.unregistered = JSON.parse(decoder.decode(metadata.unregistered))
+    }
+    console.log('Metadata event:', metadata)
```

Comment:
I personally think this should be within the SDK. Otherwise all our end-users will be copy-pasting this code for structuring the UUID appropriately and parsing the timecode/metadata :) 

@j12y wdyt?

Reply:
I'd also suggest returning `metadata.timestamp` as a `Date` object - users can print it out, or use it for sync purposes and it leaves enough room for them to use their own path.